### PR TITLE
Return int or null from all methods writing to stdout/stderr.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/TestSuite/Stub/ConsoleOutput.php [hunk 1]
```diff
@@ -59,7 +59,7 @@ public function write($message, int $newlines = 1)
             $newlines--;
         }
 
-        return false;
+        return 0;
```
Comment (markstory) [inline src/TestSuite/Stub/ConsoleOutput.php]: Should this be the length of the `$message`?

Comment (ADmad) [reply src/TestSuite/Stub/ConsoleOutput.php]: Done

File: src/TestSuite/Stub/ConsoleOutput.php [hunk 2]
```diff
@@ -59,7 +59,7 @@ public function write($message, int $newlines = 1)
             $newlines--;
         }
 
-        return false;
+        return strlen($message);
```
Comment (dereuromark) [inline src/TestSuite/Stub/ConsoleOutput.php]: strlen() can operate on arrays?

Comment (ADmad) [reply src/TestSuite/Stub/ConsoleOutput.php]: I overlooked the fact that $message can be array. Will fix it.

Returning the bytes written is necessary as it possible to overwrite a message in which case the length of previous msg is required to back track.

File: src/TestSuite/Stub/ConsoleOutput.php (hunk 1 above)
Comment (ADmad) [reply src/TestSuite/Stub/ConsoleOutput.php]: Actually returning a non zero value causes tests for `ProgressHelper` tests to fail. This is a stub class (though why is it under `src` beats me) and seems to intentionally returns empty value to make testing easier.

Comment (markstory) [reply src/TestSuite/Stub/ConsoleOutput.php]: It is in src and it is part of the integration test system.

**Review by markstory — APPROVED** [review]

**Pull request merged**